src/NServiceBus.Raw/StartableRawEndpoint.cs

```diff
@@ -32,14 +32,27 @@ public async Task<IReceivingRawEndpoint> Start()
             }
 
             var runningInstance = new RunningRawEndpointInstance(settings, receiver, transportInfrastructure, dispatcher, SubscriptionManager, localAddress);
+
             // set the started endpoint on CriticalError to pass the endpoint to the critical error action
             criticalError.SetEndpoint(runningInstance);
 
-            if (receiver != null)
+            try
             {
-                StartReceiver(receiver);
+                // set the started endpoint on CriticalError to pass the endpoint to the critical error action
+                criticalError.SetEndpoint(runningInstance);
+
+                if (receiver != null)
+                {
+                    StartReceiver(receiver);
+                }
+
+                return runningInstance;
+            }
+            catch
+            {
+                await runningInstance.Stop();
```

Comment:
Should we mark the instance somehow as failed to ensure any subsequent attempts to `Start` or `Stop` throw exception?

Reply:
We ran a test and found that calling Start again after this method throws will result in a "KeyNotFound" exception instead of the AMQP 404 it was throwing before.  How might we go about flagging the instance as failed?  We considered adding a key into the `settings` object to indicate the endpoint instance is in a bad state.  That or a `bool` class member we can check before attempting to start again similar to what the receiver does.

Reply:
I think a bool flag in the `StartableRawEndpoint` would be enough to prevent subsequent attempts at starting it.

Reply:
Done.